Mathlib/RingTheory/PowerSeries/Order.lean

```diff
@@ -157,11 +162,10 @@ private theorem order_add_of_order_eq.aux (φ ψ : R⟦X⟧) (_h : order φ ≠
 theorem order_add_of_order_eq (φ ψ : R⟦X⟧) (h : order φ ≠ order ψ) :
     order (φ + ψ) = order φ ⊓ order ψ := by
   refine le_antisymm ?_ (min_order_le_order_add _ _)
-  by_cases H₁ : order φ < order ψ
-  · apply order_add_of_order_eq.aux _ _ h H₁
-  by_cases H₂ : order ψ < order φ
-  · simpa only [add_comm, inf_comm] using order_add_of_order_eq.aux _ _ h.symm H₂
-  exfalso; exact h (le_antisymm (not_lt.1 H₂) (not_lt.1 H₁))
+  rcases lt_trichotomy φ.order ψ.order with (φ_lt_ψ | φ_eq_ψ | ψ_lt_φ)
```

Comment:
You could do `h.lt_or_lt` here so that you don't need to handle the equality case